src/pepr/operator/controllers/network/policies.ts

```diff
@@ -0,0 +1,102 @@
+import { K8s, Log, kind } from "pepr";
+
+import { Allow, Direction, Gateway, UDSPackage, getOwnerRef } from "../../crd";
+import { sanitizeResourceName } from "../utils";
+import { allowEgressDNS } from "./defaults/allow-egress-dns";
+import { allowEgressIstiod } from "./defaults/allow-egress-istiod";
+import { allowIngressSidecarMonitoring } from "./defaults/allow-ingress-sidecar-monitoring";
+import { defaultDenyAll } from "./defaults/default-deny-all";
+import { generate } from "./generate";
+
+export async function networkPolicies(pkg: UDSPackage, namespace: string) {
+  const customPolicies = pkg.spec?.network?.allow ?? [];
+  const pkgName = pkg.metadata!.name!;
+
+  // Get the current generation of the package
+  const generation = (pkg.metadata?.generation ?? 0).toString();
+
+  const policies = [
+    // All traffic must be explicitly allowed
+    defaultDenyAll(namespace),
+
+    // Allow DNS lookups
+    allowEgressDNS(namespace),
+
+    // Istio rules
+    allowEgressIstiod(namespace),
+    allowIngressSidecarMonitoring(namespace),
+  ];
+
+  // Process custom policies
+  for (const policy of customPolicies) {
+    const generatedPolicy = generate(namespace, policy);
+    policies.push(generatedPolicy);
+  }
+
+  // Generate NetworkPolicies for any VirtualServices that are generated
+  const exposeList = pkg.spec?.network?.expose ?? [];
+  for (const expose of exposeList) {
+    const { gateway = Gateway.Tenant, port, podLabels, targetPort } = expose;
+
+    // Create the NetworkPolicy for the VirtualService
+    const policy: Allow = {
+      direction: Direction.Ingress,
+      podLabels,
+      remoteNamespace: `istio-${gateway}-gateway`,
+      remotePodLabels: {
+        app: `${gateway}-ingressgateway`,
+      },
+      // Use the same port as the VirtualService if targetPort is not set
+      port: targetPort ?? port,
```

Comment:
```suggestion
      port: targetPort ?? port,
      description: `${Object.values(podLabels)} Istio ${gateway} gateway`,
```
I think this makes these istio ones a bit more clean without causing any collisions? Examples:
- `allow-neuvector-ingress-neuvector-manager-pod-istio-admin-gateway`
- `allow-grafana-ingress-grafana-istio-admin-gateway`

Only real change is it drops the namespace for the gateway (only ever one gateway per namespace, so no conflicts caused by this). I think that's the best we can do for this one without causing name conflicts, and wouldn't want to risk that here since these are generated (no option to add description via the package CR).